__tests__/api.test.js

```diff
@@ -161,6 +161,70 @@ describe("GET /api/articles", () => {
             })
         })
     })
+    test("Checking that the articles are sorted in descending order based on the date", () => {
+        
+        return request(app).get("/api/articles").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toBeSortedBy("created_at", {descending : true});
+        })
+    })
+    test("200 - Adding 'sort_by' inside the query", () => {
+        return request(app).get("/api/articles?sort_by=comment_count").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toBeSortedBy("comment_count", {descending: true, coerce: true})
+        })
+    })
+    test("200 - Adding 'order' inside the query", () => {
+        return request(app).get("/api/articles?sort_by=votes&order=ASC").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toBeSortedBy("votes");
+        })
+    })
+    test("200 - Adding 'topic' inside the query", () => {
+        return request(app).get("/api/articles?topic=cats").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toHaveLength(1);
+            articles.forEach((article) => {
+                expect(article.topic).toBe("cats");
+            })
+        })
+    })
+    test("200 - Testing complete query", () => {
+        return request(app).get("/api/articles?topic=mitch&sort_by=votes&order=ASC").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toHaveLength(11);
+            expect(articles).toBeSortedBy("votes");
+            articles.forEach((article) => {
+                expect(article.topic).toBe("mitch");
+            })
+        })
+    })
+    test("200 - Topic exists but has no articles", () => {
+        return request(app).get("/api/articles?topic=paper").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toEqual([]);
+        })
+    })
+    test("400 - Setting invalid sort_by value", () => {
+        return request(app).get("/api/articles?sort_by=apples").expect(400).then((res) => {
+            const {message} = res.body;
+            expect(message).toBe("Invalid sort_by value");
+        })
+    })
+    test("400 - Setting invalid topic value", () => {
+
+        return request(app).get("/api/articles?topic=apples").expect(400).then((res) => {
```

Comment:
This test for topic=notAnExistingTopic was correct as a 404 status as it was valid syntax wise but not found, need another test for 400 status when topic=234525 as topics are non-numeric